src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -17,11 +17,14 @@ public abstract class FunctionInvokerBase : IFunctionInvoker, IDisposable
         private FileSystemWatcher _fileWatcher;
         private bool _disposed = false;
 
-        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata)
+        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata, ITraceWriterFactory traceWriterFactory = null)
         {
             Host = host;
             Metadata = functionMetadata;
-            TraceWriter = CreateTraceWriter(host.ScriptConfig, functionMetadata.Name);
+
+            traceWriterFactory = traceWriterFactory ?? new FunctionTraceWriterFactory(functionMetadata.Name, Host.ScriptConfig);
+            TraceWriter = traceWriterFactory.Create();
+            PrimaryHostTraceWriter = new ConditionalTraceWriter(TraceWriter, () => Host.IsPrimary);
```

Comment:
We can. I actually had an implementation that did something similar. I'll send an update with that.